Test Controller

robot269006 opened the pull request:
# WHAT
Tests for messages_controller

# WHY
To check if the instances are working as intended, and to check if the view files are rendered as intended

robot269006 committed:
Pre-settings

1. Create controller_spec file
2. Comment out before_action on controller.rb

robot269006 committed:
Enable user login (devise and macros setting)

1. Add config with devise and controller macros
2. Controller macros setting
3. Test “login_user” method defined in macros
4. Password settings as login_user method showed error indicating that
the password is not set

robot269006 committed:
Controller Test (GET #index render)

1. Fix password settings -> it caused NameError when running rake
routes, since it was coded: pass = Faker::etc
2. Test render -> NoMethod Error, saying that rails-controller-testing
gem has to be installed
3. Add rails-controller-testing to gemfile, and install
4. Add group, message, user instances

robot269006 committed:
Controller Test (GET #index instances)

1. Create list of factory instances by “create_list” method
2. Use “sort” method to sort the instances order within the messages
array

robot269006 committed:
Controller Test (POST #create render and instances)

1. Apply “Context” to describe save success and failure
2. Set redirect (and fixed render settings in /groups/index)
3. Set rendering when saving message fails
4. Set instance test -> only needs testing for Message.new

robot269006 committed:
Put back the before_action

put back the before_action in messages_controller

robot269006 commented:
<img width="533" alt="2017-05-09 23 10 45" src="https://cloud.githubusercontent.com/assets/24761206/25854967/cd0599b4-350c-11e7-9144-2c485e1bd761.png">

k0mach1 commented on spec/controllers/message_controller_spec.rb:
```diff
@@ -0,0 +1,47 @@
+    let(:group)   { create(:group) }
+    let(:message) { create(:message) }
+    let(:user)    { create(:user) }
+    let(:message_params) do
```
message_paramsは必要な文脈でのみ使われる様に場所を変えてください。

robot269006 replied:
ありがとうございます、修正させていただきました。

k0mach1 commented on spec/controllers/message_controller_spec.rb:
```diff
@@ -0,0 +1,47 @@
+      it 'assigns @messages instance correctly' do
+        messages = create_list(:message, 5, group_id: group.id)
+        get :index, params: { group_id: group.id}
+        expect(assigns(:messages)).to match(messages.sort{ |a, b| a.created_at <=> b.created_at })
```
この行はもう少しシンプルに書けます

robot269006 replied:
.sort以降の表現の部分をシンプルにしてみましたが、ちょっと間違ってますか。。？

k0mach1 reviewed (commented):
コメントいたしました。

robot269006 committed:
Revision (controller_spec edit)

1. Move :message_params to POST#create division
2. Correct redirect root_path -> group_messages_path
3. Shortened “.sort” to “sort_by”

k0mach1 commented:
LGTMです！
